crates/ruff_formatter/src/builders.rs

```diff
@@ -2534,17 +2534,17 @@ impl<'a, Context> BestFitting<'a, Context> {
 
 impl<Context> Format<Context> for BestFitting<'_, Context> {
     fn fmt(&self, f: &mut Formatter<Context>) -> FormatResult<()> {
-        let mut buffer = VecBuffer::new(f.state_mut());
         let variants = self.variants.items();
 
         let mut formatted_variants = Vec::with_capacity(variants.len());
 
         for variant in variants {
+            let mut buffer = VecBuffer::with_capacity(8, f.state_mut());
```

Comment:
Let's always initialize the vector with a non-zero capacity because it always writes at least 3 entries, and lets double it to avoid resizing.